### PR TITLE
concrete-product endpoints for b2b

### PR DESCRIPTION
**Pull request by AnastasiiaYefremkinaa**
Added tests for
- concrete-product-availabilities
- concrete-product-image-sets
- concrete-product-prices
- concrete-products

### CODE REVIEW

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    And Response should return error message:    Can`t find concrete product image sets.
+
+Request_product_image_with_empty_SKU
+    When I send a GET request:    /concrete-products//concrete-product-image-sets
+    Then Response status code should be:    404
+    And Response reason should be:    Not Found
+    And Response should return error code:    302
+    And Response should return error message:    Concrete product is not found.
+
+Request_product_image_with_error_in_request
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot:20]: remove as not needed

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    And Response should return error message:     Concrete product is not found.
+
+Request_product_image_by_concrete_sku_product_doesn't_exist
+    When I send a GET request:    /concrete-products/4567890/concrete-product-image-sets
+    Then Response status code should be:    404
+    And Response reason should be:    Not Found
+    And Response should return error code:    304
+    And Response should return error message:     Can`t find concrete product image sets.
+
+Request_product_image_using_wrong_method_POST
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot:40]: remove all such tests here

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+Request_product_prices_with_empty_SKU
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token} 
+    When I send a GET request:    /concrete-products//concrete-product-prices
+    Then Response status code should be:    400
+    And Response reason should be:    Bad Request
+    And Response should return error code:    312
+    And Response should return error message:    Concrete product sku is not specified.
+
+Request_product_prices_with_error_in_request
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot:24]: remove

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+Request_product_prices_by_concrete_sku_product_doesn't_exist
+    [Setup]    Run Keywords    I get access token for the customer:    ${yves_user_email}
+    ...    AND    I set Headers:    Authorization=${token}  
+    When I send a GET request:    /concrete-products/4567890/concrete-product-prices
+    Then Response status code should be:    404
+    And Response reason should be:    Not Found
+    And Response should return error code:    308
+    And Response should return error message:     Can`t find concrete product prices.
+
+Request_product_prices_using_wrong_method_POST
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot:48]: remove all bellow

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot [hunk 5]
```diff
@@ -0,0 +1,70 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot:4]: tests for getting prices with missing and invalid token are missing, please add these tests

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+
+Request_product_concrete_with_empty_SKU
+    When I send a GET request:    /concrete-products/   
+    Then Response status code should be:    400
+    And Response reason should be:    Bad Request
+    And Response should return error code:    312
+    And Response should return error message:    Concrete product sku is not specified.
+
+Request_product_concrete_with_error_in_request
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot:29]: remove

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    And Response should return error message:     Not Found
+
+Request_product_concrete_with_special_characters
+    When I send a GET request:    /concrete-products/~!@#$%^&*()_+/
+    Then Response status code should be:    404
+    And Response reason should be:    Not Found
+    And Response should return error code:    302
+    And Response should return error message:     Concrete product is not found.
+
+Request_product_concrete_using_wrong_method_POST
```
Comment (OKrivtsova) [inline tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot:42]: remove all below

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/negative.robot [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    And Response should return error message:    Availability is not found.
+
+Request_concrete_availability_with_missing_concrete_SKU
+    When I send a GET request:    /concrete-products//concrete-product-availabilities
+    Then Response status code should be:    400
+    And Response reason should be:   Bad Request
+    And Response should return error code:    312
+    And Response should return error message:    Concrete product sku is not specified.
+
+Request_concrete_availability_by_sku_with_error_in_request
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/negative.robot:27]: removed

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/negative.robot [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    And Response should return error message:    Not Found
+
+Request_concrete_availability_by_special_characters
+    When I send a GET request:    /concrete-products/±!@#$%^&*()/concrete-product-availabilities
+    Then Response status code should be:    404
+    And Response reason should be:   Not Found
+    And Response should return error code:    302
+    And Response should return error message:    Concrete product is not found.
+
+Request_concrete_availability_by_using_wrong_method_POST
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/negative.robot:40]: removed all such tests

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot (hunk 1 above)
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot:20]: removed

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot (hunk 2 above)
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_image_sets/negative.robot:40]: removed all such tests

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot (hunk 3 above)
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot:24]: removed

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot (hunk 4 above)
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_prices/negative.robot:48]: removed all such tests

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot (hunk 6 above)
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot:29]: removed

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot (hunk 7 above)
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_products/negative.robot:42]: removed all such tests

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/positive.robot [hunk 10]
```diff
@@ -0,0 +1,29 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
+*** Test Cases ***
+Request_concrete_availability_by_concrete_SKU_with_stock
+    When I send a GET request:    /concrete-products/${concrete_product_with_alternative_sku}/concrete-product-availabilities
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/positive.robot:7]: added new variable

File: tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/positive.robot [hunk 11]
```diff
@@ -0,0 +1,29 @@
+*** Settings ***
+Suite Setup       SuiteSetup
+Resource    ../../../../../../resources/common/common_api.robot
+
```
[the remark this answers is not in the document]
Comment (AnastasiiaYefremkinaa) [reply tests/api/b2b/glue/concrete_products_endpoints/concrete_product_availabilities/positive.robot:4]: done. added test, created a ticket, and spoke to PO: https://spryker.atlassian.net/browse/CC-15984

**Pull request merged**